packages/opencensus-exporter-trace-stackdriver/src/stackdriver.ts

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {classes, logger, types} from '@opencensus/opencensus-core';
+import {auth, JWT} from 'google-auth-library';
+import {google} from 'googleapis';
+
+google.options({headers: {'x-opencensus-outgoing-request': 0x1}});
+const cloudTrace = google.cloudtrace('v1');
+
+type ExporterBuffer = typeof classes.ExporterBuffer;
+
+/**
+ * Options for stackdriver configuration
+ */
+export interface StackdriverExporterOptions extends types.ExporterConfig {
+  /**
+   * projectId project id defined to stackdriver
+   */
+  projectId: string;
+}
+
+interface TracesWithCredentials {
+  projectId: string;
+  resource: {traces: {}};
+  auth: JWT;
+}
+
+/** Format and sends span information to Stackdriver */
+export class StackdriverTraceExporter implements types.Exporter {
+  projectId: string;
+  exporterBuffer: classes.ExporterBuffer;
+  logger: types.Logger;
+  failBuffer: types.SpanContext[] = [];
+
+  constructor(options: StackdriverExporterOptions) {
+    this.projectId = options.projectId;
+    this.logger = options.logger || logger.logger();
+    this.exporterBuffer = new classes.ExporterBuffer(this, options);
+  }
+
+  /**
+   * Is called whenever a span is ended.
+   * @param root the ended span
+   */
+  onEndSpan(root: types.RootSpan) {
+    this.exporterBuffer.addToBuffer(root);
+  }
+
+  /**
+   * Publishes a list of root spans to Stackdriver.
+   * @param rootSpans
+   */
+  publish(rootSpans: types.RootSpan[]) {
```

Comment:
The base class seems to return `void`?

Reply:
TODO let's review the base class in future reviews.